test/test_ppca.py

```diff
@@ -31,21 +44,22 @@ def test_probabilistic_pca(x_ng, minibatch, ppca_flavor, learn_mean):
     else:
         x_mean_g = x_ng.mean(axis=0)
 
+    # dataloader
+    batch_size = n // 2 if minibatch else n
+    train_loader = DataLoader(
+        dataset(x_ng),
+        batch_size=batch_size,
+    )
+    # model
     pyro.clear_param_store()
     ppca = ProbabilisticPCAPyroModule(
         n_cells=n, g_genes=g, k_components=k, ppca_flavor=ppca_flavor, mean_g=x_mean_g
     )
-    elbo = infer.Trace_ELBO()
-    adam = optim.Adam({"lr": 1e-2})
-    svi = infer.SVI(ppca.model, ppca.guide, adam, elbo)
-    for i in range(5000):
-        if minibatch:
-            batch_size = n // 2
-            ind = torch.randint(n, (batch_size,))
-            x_batch = x_ng[ind]
-        else:
-            x_batch = x_ng
-        svi.step(x_batch)
+    training_plan = PyroTrainingPlan(ppca, optim_kwargs={"lr": 1e-2})
+    # trainer
+    trainer = pl.Trainer(accelerator="cpu", max_epochs=4000)
```

Comment:
Out of curiosity, does PL _control_ the optimizer? e.g. if one wants to user a scheduler, would PL automatically give the optimizer to the scheduler?

Reply:
Not sure, I will have to dig more into it to see how it is handled.